### DISCUSSION TITLE
How to solve the question that "the list of modifications might is empty if the commit is a merge commit"

### DISCUSSION THREAD

**Discussion by cp3-three** [ishepard/pydriller#244] [Q&A]
I noticed that "the list of modifications might is empty if the commit is a merge commit"，but I want to know which methods was modified in a merge commit that compare to "parent[0]" . I wonder if there is any other way I can do it. Thank you very much if you can get an answer!

**Comment by ishepard** [comment]
Most of the time a merge commit doesn't have any modified file, so the list is always empty. 
The only exception to that are merge conflicts. In this case the files are modified in the merge commit.

I have a detailed answer (with examples) to this question here: https://github.com/ishepard/pydriller/issues/89#issuecomment-590243707

**Reply by cp3-three** [reply]
> Most of the time a merge commit doesn't have any modified file, so the list is always empty. The only exception to that are merge conflicts. In this case the files are modified in the merge commit.
> 
> I have a detailed answer (with examples) to this question here: [#89 (comment)](https://github.com/ishepard/pydriller/issues/89#issuecomment-590243707)

I know about that. But there's a need in my project.
![image](https://user-images.githubusercontent.com/52562731/215440099-9ee1a9de-3b9f-4864-ae9e-aad18ff16e55.png)
I want to compare commit5[merge] and commit3[parent0] . So I can know what code is modified for commit2 and commit4 in total.
 I wonder if there is any other way I can do it.

**Comment by ishepard** [comment]
So you want "all" code modified in a branch. 2 options:
- there is no quick way of doing it with pydriller, the only thing that comes in my mind is to iterate over the entire branch and save the code modifications
- you can try doing it manually with Gitpython. The commit object has a parameter called _c_object, you can [see it here](https://github.com/ishepard/pydriller/blob/master/pydriller/domain/commit.py#L508). That is the commit of GitPython. Then you can try to do something similar to git diff parent[0], something like [I do here](https://github.com/ishepard/pydriller/blob/master/pydriller/domain/commit.py#L694).
